### ISSUE TITLE
Deduplicate zcl.json with extension data

### ISSUE THREAD

**Issue by andy31415** [project-chip/connectedhomeip#19319]
zcl.json and zcl-with-test-extensions.json need to contain practically the same data except extensions contains one more path.

This duplication is painful to maintain. Instead we should dedup, however this would ideally be done once zap supports multiple --zcl arguments.

Originating from:

As a followup: please add a lint that this file and zcl.json are in fact identical except for whatever the intended differences are.  Otherwise they're going to end up diverging.

_Originally posted by @bzbarsky-apple in https://github.com/project-chip/connectedhomeip/pull/19274#discussion_r892491152_

**Comment by bzbarsky-apple** [comment]
I don't think we should wait on ZAP here.  Something using `diff -u0 file1 file2 | tail +4` and comparing that to a known string should work....
